src/util/string.cpp

```diff
@@ -887,3 +887,18 @@ std::string sanitizeDirName(const std::string &str, const std::string &optional_
 
 	return wide_to_utf8(safe_name);
 }
+
+
+void safe_print_string(std::ostream &os, const std::string &str)
+{
+	std::ostream::fmtflags flags = os.flags();
+	os << std::hex;
+	for (const char c : str) {
+		if (IS_ASCII_PRINTABLE_CHAR(c) || c == '\n' || c == '\t') {
```

Comment:
so no utf-8? :thinking: 
what about
```suggestion
		if ((c >= 0x20 && c <= 0xf5) || c == '\n' || c == '\t') {
```

Reply:
I'd like to allow full Unicode, but UTF-8 has some control characters that could cause issues, so I'm not sure if it makes sense to allow them when we're blocking other control characters.  If we want to allow that we could (although the output stream might not be using a UTF-8 encoding).

Reply:
ASCII control characters can change the terminal state (on Unix-like OSs) while Unicode ones are just visual, so should be fine to allow those IMO.

Reply:
I added a commit to allow UTF-8.  It uses the UTF-8 macros though, so DEL and bytes that aren't UTF-8 start or inner bytes aren't allowed.